src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -317,17 +313,7 @@ public char PeekChar(int delta)
             int position = this.Position;
             this.AdvanceChar(delta);
 
-            char ch;
-            if (_offset >= _characterWindowCount
-                && !MoreChars())
-            {
-                ch = InvalidCharacter;
-            }
-            else
-            {
-                // N.B. MoreChars may update the offset.
-                ch = _characterWindow[_offset];
-            }
+            char ch = PeekChar();
```

Comment:
You could always move it to an intermediate (internal) method and put an `AgressiveInlining` hint on it. Wouldn't guarantee the inlining, but it should work for this case.